app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt

```diff
@@ -309,27 +309,32 @@ class CreateRecipeViewModel(
                   C.Tag.FIRESTORE_RECIPE_IMAGE_NAME,
                   ImageDirectoryType.RECIPE,
                   onSuccess = { uri ->
-
-                    // Set the URL to the Builder
-                    val url = uri.toString()
-                    recipeBuilder.setUrl(url)
-
-                    // Build the Recipe
-                    val recipe = recipeBuilder.build()
-
-                    // Add the Recipe to the Repository
-                    repository.addRecipe(
-                        recipe,
-                        onSuccess = {
-                          onSuccess(recipe)
-                          _publishStatus.value = RECIPE_PUBLISHED_SUCCESS_MESSAGE
-                          recipeBuilder.clear()
-                        },
-                        onFailure = { exception ->
-                          _publishStatus.value =
-                              RECIPE_PUBLISH_ERROR_MESSAGE.format(exception.message)
-                          onFailure(exception)
-                        })
+                    try {
+
```

Comment:
I believe the `try-catch` block might not be necessary here, as these two lines don't seem to throw any exceptions. Removing it could help streamline the code. What do you think?

```kotlin  
val url = uri.toString()
recipeBuilder.setUrl(url)
````
Btw to you can add this test to cover the failure (Try to build without ingredient): 
```kotlin
@Test
    fun `test publishRecipe with incorrect recipe field throw error`() = runTest {
        val defaultRecipe = createDefaultRecipe()
        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)

        createRecipeViewModel.updateRecipeName("Chocolat") // Blank name
        createRecipeViewModel.updateRecipeInstructions("ww www www")
        createRecipeViewModel.setBitmap(bitmap, 90)

        `when`(mockRepository.getNewUid()).thenReturn(defaultRecipe.uid)
        createRecipeViewModel.setBitmap(bitmap, 90)
        // Define the onFailure callback to validate if it was called correctly
        var onFailureCalled = false
        createRecipeViewModel.publishRecipe(
            onSuccess = { fail("Expected onFailure, but onSuccess was called instead.") },
            onFailure = { errorMessage ->
                onFailureCalled = true
                assertEquals("At least one ingredient is required.", errorMessage.message)
            })

        advanceUntilIdle()

        `when`(
            mockImageRepository.uploadImage(
                any(), any(), any(), any(), onSuccess = any(), onFailure = any()))
            .thenAnswer { invocation ->
                val onSuccessCallback = invocation.arguments[4] as () -> Unit
                onSuccessCallback()
            }
        `when`(mockRepository.getNewUid()).thenReturn(defaultRecipe.uid)

        `when`(
            mockImageRepository.getImageUrl(
                any(), any(), any(), onSuccess = any(), onFailure = any()))
            .thenAnswer { invocation ->
                val onSuccessCallback = invocation.arguments[3] as (Uri) -> Unit
                onSuccessCallback(Uri.EMPTY)
            }

        createRecipeViewModel.publishRecipe(onSuccess = {}, onFailure = {})
        assertEquals(
            "At least one ingredient is required.", createRecipeViewModel.publishStatus.value)
    }
```